src/algebra/ordered_monoid.lean

```diff
@@ -396,6 +396,29 @@ instance [add_comm_monoid α] : add_comm_monoid (with_top α) :=
   ..@additive.add_comm_monoid _ $ @comm_monoid_with_zero.to_comm_monoid _ $
     @with_zero.comm_monoid_with_zero (multiplicative α) _ }
 
+section sub_neg
+
+variable [add_group α]
+
+instance : sub_neg_monoid (with_top α) :=
+{ neg := option.map (λ (a : α), -a),
```

Comment:
I think providing `sub` explicitly might be a good idea, as something like
```suggestion
{ neg := option.map (λ (a : α), -a),
  sub := λ o₁ o₂, o₁.bind (λ a, o₂.map (λ b, a - b)),
  sub_eq_add_neg := sorry,
```

Reply:
I can add it, no problem. What is the reasoning though, is the `a + -b` default implementation somehow not good?

Reply:
If the user has chosen to make `has_sub A` not defeq to add/neg, then presumably their reason transfers to `has_sub (with_top A)`.